File: cpp/1193.cpp
#include <iostream>
#include <string>
#include <bitset>

using namespace std;

int main()
{
    int n;
    cin >> n;

    for (int i = 1; i <= n; i++) {
        int number, j = 0;
        string num, basis, result;
        cin >> num >> basis;
        cout << "Case " << dec << i << ":\n";

        if (basis == "bin") {
            number = stoi(num, nullptr, 2);
            cout << dec << number << " dec\n";
            cout << hex << number << " hex\n";

        } else if (basis == "hex") {
            number = stoi(num, nullptr, 16);
            result = bitset<32>(number).to_string();
            while (result[j] == '0') j++;
            result.erase(0, j);
            cout << dec << number << " dec\n";
            cout << result << " bin\n";

        } else {
            number = stoi(num);
            result = bitset<32>(number).to_string();
            while (result[j] == '0') j++;
            result.erase(0, j);
            cout << hex << number << " hex\n";
            cout << result << " bin\n";
        }

        cout << '\n';
    }

    return 0;
}

File: py/1193.py
def conversion():
    n = int(input())

    for i in range(1, n + 1):
        number, basis = input().split()
        print(f'Case {i}:')

        if basis == 'bin':
            print(int(number, 2), 'dec')
            print(hex(int(number, 2))[2:], 'hex')

        elif basis == 'hex':
            print(int(number, 16), 'dec')
            print(bin(int(number, 16))[2:], 'bin')

        else:
            print(hex(int(number))[2:], 'hex')
            print(bin(int(number))[2:], 'bin')

        print('')


conversion()